testdata/go/v3/memcached-operator/controllers/memcached_controller.go

```diff
@@ -103,6 +104,8 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			log.Error(err, "Failed to update Deployment", "Deployment.Namespace", found.Namespace, "Deployment.Name", found.Name)
 			return ctrl.Result{}, err
 		}
+		// Increment MemcachedNumberOfReplicasReconcileCountTotal metric by 1
+		monitoring.MemcachedNumberOfReplicasReconcileCountTotal.Inc()
```

Comment:
@assafad 

What is the goal of this metric? Check how many times the size of replicas were not == the size defined in the CR? If so we need a better name. 

However, if the goal here is let we know how much replicas it has than note that:

We only go inside here when the size is not == what is defined in the CR.
So, that means it will not count what is the total of replicas unless something goes wrong or someone updates/changes the CR. 

Therefore, I would suggest you add it at the end of the reconciliation where all operations were done suceffuly then, you can check the number of the replicas and update this metric. 

Also, why we would need a metric for this info? I think a custom metric here needs to be about something that goes wrong right?

/hold

Reply:
>What is the goal of this metric? Check how many times the size of replicas were not == the size defined in the CR? If so we need a better name.

The goal of the metric is to count how many times the number of memcached replicas was reconciled. i.e. how many times we define [found.Spec.Replicas = &size](https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v4-alpha/memcached-operator/controllers/memcached_controller.go#L100). This is why the metric's name includes `number_of_replicas_reconcile`.
Note - This metric also equals to how many times size of replicas were not == the size defined in the CR.

Reply:
As agreed, updated the metric name to `memcached_deployment_size_undesired_count_total`